packages/legacy/core/App/navigators/RootStack.tsx

```diff
@@ -1,52 +1,40 @@
-import { CardStyleInterpolators, StackCardStyleInterpolator, createStackNavigator } from '@react-navigation/stack'
-import React, { useEffect } from 'react'
+import React, { useEffect, useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
 import { DeviceEventEmitter } from 'react-native'
 
-import IconButton, { ButtonLocation } from '../components/buttons/IconButton'
 import { EventTypes } from '../constants'
 import { TOKENS, useServices } from '../container-api'
 import { useStore } from '../contexts/store'
-import { useTheme } from '../contexts/theme'
 import { useDeepLinks } from '../hooks/deep-links'
-import HistoryStack from '../modules/history/navigation/HistoryStack'
-import Chat from '../screens/Chat'
 import { BifoldError } from '../types/error'
-import { Screens, Stacks, TabStacks } from '../types/navigators'
-import { testIdWithKey } from '../utils/testable'
-
-import ConnectStack from './ConnectStack'
-import ContactStack from './ContactStack'
-import DeliveryStack from './DeliveryStack'
-import NotificationStack from './NotificationStack'
-import ProofRequestStack from './ProofRequestStack'
-import SettingStack from './SettingStack'
-import TabStack from './TabStack'
-import { useDefaultStackOptions } from './defaultStackOptions'
-import CredentialDetails from '../screens/CredentialDetails'
-import OpenIDCredentialDetails from '../modules/openid/screens/OpenIDCredentialDetails'
+import MainStack from './MainStack'
 
 const RootStack: React.FC = () => {
   const [store, dispatch] = useStore()
   const { agent } = useAgent()
   const { t } = useTranslation()
-  const theme = useTheme()
-  const defaultStackOptions = useDefaultStackOptions(theme)
-  const [splash, OnboardingStack, CustomNavStack1, ScreenOptionsDictionary, loadState] = useServices([
-    TOKENS.SCREEN_SPLASH,
-    TOKENS.STACK_ONBOARDING,
-    TOKENS.CUSTOM_NAV_STACK_1,
-    TOKENS.OBJECT_SCREEN_CONFIG,
-    TOKENS.LOAD_STATE,
-  ])
+  const [OnboardingStack, loadState] = useServices([TOKENS.STACK_ONBOARDING, TOKENS.LOAD_STATE])
+  // remove connection on mobile verifier proofs if proof is rejected
+  // regardless of if it has been opened
+  const declinedProofs = useProofByState([ProofState.Declined, ProofState.Abandoned])
 
   useDeepLinks()
 
-  // remove connection on mobile verifier proofs if proof is rejected regardless of if it has been opened
-  const declinedProofs = useProofByState([ProofState.Declined, ProofState.Abandoned])
+  const isAuthenticated = useMemo(
+    () => store.authentication.didAuthenticate && store.onboarding.postAuthScreens.length === 0,
+    [store.authentication.didAuthenticate, store.onboarding.postAuthScreens]
+  )
+
+  const isOnboardingComplete = useMemo(
+    () =>
+      (store.onboarding.onboardingVersion !== 0 && store.onboarding.didCompleteOnboarding) ||
+      (store.onboarding.onboardingVersion === 0 && store.onboarding.didConsiderBiometry),
+    [store.onboarding]
```

Comment:
Might be better to watch the specific properties rather than the onboarding object 

```suggestion
    [store.onboarding.onboardingVersion, store.onboarding.didCompleteOnboarding, store.onboarding.didConsiderBiometry]
```